Adds Learn CSS Animation (#5238)
GoogleChrome/web.dev: src/site/content/en/learn/css/animations/index.md

along a timeline.

Let's use the "pulser" as a context for this.
The entire animation runs for 1 second and runs over 2 states.

{% Img
src="image/VbAJIREinuYvovrBzzvEyZOpw5w1/pWCGpgzJqJNTBluK5u8S.svg",
alt="The states of the pulser animation over the 1 second timeframe",
width="800",
height="340" %}

There's a specific point where each of these animation states start and end.
You map these out on the timeline with keyframes.

{% Img
src="image/VbAJIREinuYvovrBzzvEyZOpw5w1/wmKVq5GFj7cf6jfuLiT8.svg",
alt="The same diagram as before, but this time, with keyframes",
width="800",
height="440" %}

### `@keyframes`

Now you know what a keyframe is,
that knowledge should help you understand how the CSS
[`@keyframes`](https://developer.mozilla.org/en-US/docs/Web/CSS/@keyframes) rule works.
Here is a basic rule with two states.

```css
@keyframes my-animation {
	from {
		transform: translateY(20px);
	}
	to {
		transform: translateY(0px);
	}
}
```

The first part to note is the
[custom ident](https://developer.mozilla.org/en-US/docs/Web/CSS/custom-ident)
(custom identifier)—or in more human terms, the name of the keyframes rule.
This rule's identifier is `my-animation`.

> The custom ident works like a function name. Which, as you learned in the functions lesson,

Reviewer: ```suggestion
The custom identifier works like a function name. Which, as you learned in the [functions module](/learn/css/functions),
```

lets you reference the keyframes rule elsewhere in your CSS code.

{% Aside %}

> A custom-ident is used in various places in CSS,

Reviewer: ```suggestion
A `<custom-ident>` is used in various places in CSS,
```

- `reverse`: runs backwards over your timeline
- `alternate`: for each animation iteration, the timeline will run forwards or backwards in sequence
- `alternate-reverse`: the reverse of `alternate`

### `animation-delay`

```css
.my-element {
	animation-delay: 5s;
}
```

The [animation-delay](https://developer.mozilla.org/en-US/docs/Web/CSS/animation-delay)
property defines how long to wait before starting the animation.
Like the `animation-duration` property, this accepts a time value.

Unlike the `animation-duration` property, you *can* define this as a negative value.
If you set a negative value, the timeline in your `@keyframes` will start at that point.
For example, if your animation is 10 seconds long and you set `animation-delay` to `-5s`, it will start from half-way along your timeline.

### `animation-play-state`

```css
.my-element:hover {
	animation-play-state: paused;
}
```

The [animation-play-state](https://developer.mozilla.org/en-US/docs/Web/CSS/animation-play-state)
property allows you to play and pause the animation.
The default value is `running` and if you set it to `pause`, it will pause the animation.

{% Codepen {
  user: 'web-dot-dev',
  id: 'MWJQZyV',

>   height: 300

Reviewer: ```suggestion
  height: 400
```

- `both`: follows the rules for both `forwards` and `backwards`

### The `animation` shorthand

Instead of defining all the properties separately,
you can define them in an `animation` shorthand,
which lets you define the animation properties in the following order:

1. `animation-name`
2. `animation-duration`
3. `animation-timing-function`
4. `animation-delay`
5. `animation-iteration-count`
6. `animation-direction`
7. `animation-fill-mode`
8. `animation-play-state`

```css
.my-element {
	animation: my-animation 10s ease-in-out 1s infinite forwards forwards running;
}
```

## Considerations when working with animation

Users can define in their operating system that they prefer to reduce motion
experienced when they interact with applications and websites.
This preference can be detected using the
[prefers-reduced-motion](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
media query.

```css
@media (prefers-reduced-motion) {
  .my-autoplaying-animation {
    animation-play-state: paused;
  }
}
```

This isn't necessarily a preference of no animation,
but rather, a preference to reduce animations—

> [especially unexpected ones](https://www.smashingmagazine.com/2020/09/design-reduced-motion-sensitivities/).

Reviewer: ```suggestion
[especially unexpected ones](/prefers-reduced-motion/).
```
